src/bit_encoding/bititer.rs

```diff
@@ -225,6 +225,14 @@ impl<I: Iterator<Item = u8>> BitIter<I> {
         Ok(result_stack.pop().unwrap())
     }
 
+    /// Decode a natural number from bits.
+    ///
+    /// If a bound is specified, then the decoding terminates before trying to
+    /// decode a larger number.
+    pub fn read_natural(&mut self, bound: Option<usize>) -> Result<usize, decode::Error> {
+        decode::decode_natural(self, bound)
+    }
```

Comment:
9d942f71cce4c1c817eae8086c6adf6cd04c0ebc: We will replace `decode::decode_natural(self, bound)` with an iterative method that will probably live in `BitIter` in a future PR.